### PR TITLE
Add metrics scraping

### PR DESCRIPTION
**Pull request by jsafrane**
Configure Prometheus to collect AWS EBS CSI driver sidecar metrics.

These changes were necessary:
* For each CSI sidecar add `--http-endpoint` for metrics. Expose those metrics on localhost, because we don;'t want plain http visible from outside.
* For each CSI sidecar add another container with kube-rbac-proxy. It provides https endpoint, does authentication of Prometheus and finally proxies the metrics from the previous bullet. Each sidecar gets its own proxy!
  * Use the right ports on the node for he proxy. Since we use `hostNetwork: true`, only range 9000-9999 is available for the https endpoint.
  * Add RBAC rules to allow the proxy to create TokenReview objects to authenticate Prometheus.
* Add Service for the metrics, with an annotation to get cacert secret for it for https.
* Add RBACs to give Prometheus permissions to inspect CSI driver Services, Endpoints and Pods in the CSI driver namespace.
* Add label to Namespace to allow Prometheus checking ServiceMonitor in the namespace.
  * This must be propagated to CSO, because CSO creates the namespace!
* Finally, add ServiceMonitor to configure the scraping.

When at it, I synced `/manifests` directory with CSO.

WIP:
* wait for https://github.com/openshift/library-go/pull/1042 to get merged
* Add node metrics (if any)
* Add CSI driver metrics (if any)

### CODE REVIEW

**Comment by jsafrane** [comment]
Rebased to current library-go

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -80,6 +80,8 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 			"csidriver.yaml",
 			"controller_sa.yaml",
 			"node_sa.yaml",
+			"service.yaml",
+			"servicemonitor.yaml",
```
Comment (bertinatto) [inline pkg/operator/starter.go]: Didn't you want to use the new `WithServiceMonitorController()` instead?

https://github.com/openshift/library-go/pull/1042/files#diff-ca366a7dedcbe10d49342262a66fa7737b8205813d6bb29224aed6f828713cdeR192-R209

**Comment by bertinatto** [comment]
/hold cancel
(sorry, wrong PR)

File: pkg/operator/starter.go (hunk 1 above)
Comment (jsafrane) [reply pkg/operator/starter.go]: I think I use it. Is it some old version of the PR?

Comment (bertinatto) [reply pkg/operator/starter.go]: Seems like a caching issue

**Review by bertinatto — COMMENTED** [review]
/lgtm

**Pull request merged**